### PR TITLE
feat: Rename module to resources and look for all resources

### PR DESCRIPTION
**Pull request by rakesh-garimella**
## Description

Changes proposed in this pull request (what was done and why):

-  Rename modules to resources
- Look for all resources, dont stop at first one

Changes refer to particular issues, PRs or documents:

- https://github.com/kyma-project/telemetry-manager/issues/1388

## Traceability
- [ ] The PR is linked to a GitHub issue.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.
- [ ] If the change is user-facing, the documentation has been adjusted.
- [x] The feature is unit-tested.
- [ ] The feature is e2e-tested.

### CODE REVIEW

File: receiver/kymastatsreceiver/documentation.md [hunk 1]
```diff
@@ -24,13 +24,13 @@ The module status conditions. Possible metric values for condition status are 'T
...
-| type | The type of the module condition. | Any Str |
+| reason | The reason of the resource condition status. | Any Str |
+| status | The status of the resource condition. | Any Str |
+| type | The type of the resource condition. | Any Str |
 
-### kyma.module.status.state
+### kyma.resource.status.state
 
-The module status state, metric value is 1 for last scraped module status state, including state as metric attribute.
+The resource status state, metric value is 1 for last scraped resource status state, including state as metric attribute.
```
Comment (NHingerl) [inline receiver/kymastatsreceiver/documentation.md:33]:
```suggestion
The resource status state. Metric value is 1 for last scraped resource status state, including state as metric attribute.
```

**Review by hisarbalik — APPROVED** [review]

File: receiver/kymastatsreceiver/README.md [hunk 2]
```diff
@@ -18,7 +18,7 @@ The following settings are required:
 
 - `auth_type` (default = `serviceAccount`): Specifies the authentication method for accessing the Kubernetes API server.
    Options include `none` (no authentication), `serviceAccount` (uses the default service account token assigned to the Pod), or `kubeConfig` (uses credentials from `~/.kube/config`).
-- `modules`: A list of API group-version-resources of Kyma modules. For each group-version-resource, the status metrics are generated for the first resource instance found.
+- `resources`: A list of API group-version-resources of Kyma resources. For each group-version-resource.
```
Comment (NHingerl) [inline receiver/kymastatsreceiver/README.md:21]: the 2nd sentence seems to be missing a bit

Comment (rakesh-garimella) [reply receiver/kymastatsreceiver/README.md:21]: previously we were fetching status for only first resource. Now the behaviour has changed and we fetch for all the resources. So I removed the sentence

Comment (NHingerl) [reply receiver/kymastatsreceiver/README.md:21]: Thank you! But what does the lonely “For each group-version-resource.” mean?
Does it now belong to the previous sentence? Do you want to say
“A list of API group-version-resources of Kyma resources for each group-version-resource.”
But then, what does that mean? Is there a list of g-v-resources for Kyma resources for each g-v-resource?

**Review by NHingerl — APPROVED** [review]

**Pull request merged**